multinet/db.py

```diff
@@ -222,13 +229,38 @@ def get_workspace_metadata(name: str) -> Workspace:
 
     # Find the metadata record for the named workspace. If it's not there,
     # something went very wrong, so bail out.
-    metadata = workspace_mapping(name)
+    metadata = workspace_mapping.__wrapped__(name)
     if metadata is None:
         raise DatabaseCorrupted()
 
     return metadata
 
 
+def set_workspace_permissions(
+    name: str, permissions: WorkspacePermissions
+) -> WorkspacePermissions:
+    """Update the permissions for a given workspace."""
+    if not workspace_exists(name):
+        raise WorkspaceNotFound(name)
+
+    doc = copy.deepcopy(workspace_mapping(name))
+    if doc is None:
+        raise DatabaseCorrupted()
+
+    # TODO: Do user object validation once ORM is implemented
+
+    # To ensure owner is only changed explicitly
+    new_permissions = copy.deepcopy(permissions)
+    new_permissions["owner"] = doc["permissions"]["owner"]
+
+    doc["permissions"] = new_permissions
+    return_doc = workspace_mapping_collection().get(
+        workspace_mapping_collection().update(doc, check_rev=False)
```

Comment:
Why `check_rev=False`?

Reply:
Also, what does the `update()` method return? I'm surprised it doesn't give you back the new doc!

Reply:
Yeah, `update` only returns the metadata (so `_rev`, `_id`, etc.). The reason I added `check_rev=False`, is because I was getting a `DocumentRevisionError`, which was due to the call to `workspace_mapping` above returning an old version of the document, with an old `_rev` (since it's cached). 

Looking at this now, I actually don't like what I did, as it just ignores the problem. I'm going to push a fix that actually addresses the problem here.

Reply:
> Looking at this now, I actually don't like what I did, as it just ignores the problem. I'm going to push a fix that actually addresses the problem here.

That's why I spoke up, something felt off about setting that flag.

Reply:
6a786b4